### ISSUE TITLE
Add a way to disble cuda configurations when using generate-binary-build-matrix

### ISSUE THREAD

**Issue by seemethere** [pytorch/test-infra#744]
Specific repositories like `pytorch/text` do not build for CUDA for linux or windows so we need to have a way to disable CUDA for these specific repositories.

Instead of hardcoding which repositories need cuda we should have a variable we can pass through the re-usable workflow / script that allows us to disable cuda altogether when generating the matrix.

Usage would look like:
```yml
  generate-matrix:
    uses: ./.github/workflows/generate_binary_build_matrix.yml
    with:
      package-type: wheel
      os: linux
      with-cuda: false # <- new variable
```

**Comment by DanilBaibak** [comment]
On it!

**Comment by seemethere** [comment]
Resolved by #752
